editor/src/clj/editor/gui.clj

```diff
@@ -2806,47 +2905,59 @@
         (when (:layout user-data)
           (add-layout-options node-id user-data))))))
 
-(def node-property-fns
-  {:position [:position v4->v3]
-   :rotation [:rotation euler-v4->clj-quat]
-   :scale [:scale v4->v3]
-   :size [:manual-size v4->v3]
-   :xanchor [:x-anchor identity]
-   :yanchor [:y-anchor identity]})
-
-(defn- convert-node-desc [node-desc]
-  (into {}
-        (map (fn [[key val :as entry]]
-               (if-some [[new-key convert-fn] (node-property-fns key)]
-                 [new-key (convert-fn val)]
-                 entry)))
-        node-desc))
+(defn- node-desc->node-properties [node-desc]
+  {:pre [(map? node-desc)]} ; Gui$NodeDesc in map format.
+  (persistent!
+    (reduce (fn [node-properties [pb-field pb-value]]
+              (if-some [[prop-key pb-value->prop-value] (pb-field-to-node-property-conversions pb-field)]
+                (let [prop-value (pb-value->prop-value pb-value)]
+                  (-> node-properties
+                      (dissoc! pb-field)
+                      (assoc! prop-key prop-value)))
+                node-properties))
+            (transient node-desc)
+            node-desc)))
 
 (defn- sort-node-descs
   [node-descs]
   (let [parent-id->children (group-by :parent (remove #(str/blank? (:id %)) node-descs))
         parent->children #(parent-id->children (:id %))
-        root {:id ""}]
+        root {:id nil}]
     (rest (tree-seq parent->children parent->children root))))
 
-(def ^:private custom-gui-scene-loaders (atom (sorted-map)))
+(defonce ^:private custom-gui-scene-loaders (atom (sorted-map)))
 
 ;; SDK api
 (defn register-gui-scene-loader! [load-fn]
   (let [load-fn-sym (fn/declared-symbol load-fn)]
     (swap! custom-gui-scene-loaders assoc load-fn-sym load-fn)))
 
-;; Used by tests
-(defn clear-custom-gui-scene-loaders-for-tests! []
-  (reset! custom-gui-scene-loaders (sorted-map)))
-
```

Comment:
Moved to bottom of file. Now also clears `custom-node-type-infos`, encountered later.